src/App.tsx

```diff
@@ -1,27 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAllGoods, get5FirstGoods, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const getGoodsFromServer = (getProducts: Promise<Good[]>) => {
+    getProducts.then(goods => setVisibleGoods(goods));
```

Comment:
```suggestion
    getProducts.then(setVisibleGoods);
```
it will work this way